src/main/java/teammates/common/datatransfer/questions/FeedbackMsqQuestionDetails.java

```diff
@@ -417,61 +444,66 @@ public String getQuestionResultStatisticsHtml(List<FeedbackResponseAttributes> r
+    }
+
+    private int getNumOfChoicesSelected(FeedbackResponseAttributes response, int numChoicesSelected,
+            Map<String, Integer> answerFrequency) {
+        int numChoices = 0;
+        List<String> answerStrings =
+                ((FeedbackMsqResponseDetails) response.getResponseDetails()).getAnswerStrings();
+        boolean isOtherOptionAnswer =
+                ((FeedbackMsqResponseDetails) response.getResponseDetails()).isOtherOptionAnswer();
+        String otherAnswer = "";
+
+        if (isOtherOptionAnswer) {
+            if (!answerFrequency.containsKey("Other")) {
+                answerFrequency.put("Other", 0);
             }
 
-            for (String answerString : answerStrings) {
-                if (answerString.isEmpty()) {
-                    continue;
-                }
+            answerFrequency.put("Other", answerFrequency.get("Other") + 1);
 
-                isContainsNonEmptyResponse = true;
-                numChoicesSelected++;
+            numChoices++;
+            // remove other answer temporarily to calculate stats for other
+            // options
+            otherAnswer = answerStrings.get(answerStrings.size() - 1);
+            answerStrings.remove(otherAnswer);
+        }
 
-                if (!answerFrequency.containsKey(answerString)) {
-                    answerFrequency.put(answerString, 0);
-                }
-                answerFrequency.put(answerString, answerFrequency.get(answerString) + 1);
+        for (String answerString : answerStrings) {
+            if (answerString.isEmpty()) {
+                continue;
             }
+            isContainsNonEmptyResponse = true;
+            numChoices++;
 
-            // restore other answer if any
-            if (isOtherOptionAnswer) {
-                answerStrings.add(otherAnswer);
+            if (!answerFrequency.containsKey(answerString)) {
+                answerFrequency.put(answerString, 0);
             }
+            answerFrequency.put(answerString, answerFrequency.get(answerString) + 1);
         }
 
-        if (!isContainsNonEmptyResponse) {
-            return "";
+        // restore other answer if any
+        if (isOtherOptionAnswer) {
+            answerStrings.add(otherAnswer);
         }
 
-        DecimalFormat df = new DecimalFormat("#.##");
-
-        StringBuilder fragments = new StringBuilder();
-        for (Entry<String, Integer> entry : answerFrequency.entrySet()) {
-            fragments.append(Templates.populateTemplate(FormTemplates.MCQ_RESULT_STATS_OPTIONFRAGMENT,
-                                Slots.MCQ_CHOICE_VALUE, entry.getKey(),
-                                Slots.COUNT, entry.getValue().toString(),
-                                Slots.PERCENTAGE, df.format(100 * (double) entry.getValue() / numChoicesSelected)));
+        return numChoices + numChoicesSelected;
+    }
 
+    protected double checkForNaN(int numChoice, double entryValue) {
```

Comment:
This method should not be needed since if numChoice is zero, it would imply there are no responses to the question. Should consider reordering the parameters and renaming the method (maybe like `divideOrReturnZero`) so that it is more intuitive if you think there is still a need for this method.